sqlmesh/core/model/definition.py

```diff
@@ -1170,7 +1170,9 @@ def render(
             for column in date_or_time_columns:
                 df[column] = pd.to_datetime(df[column])
             df[bool_columns] = df[bool_columns].apply(lambda i: str_to_bool(str(i)))
-            df[string_columns] = df[string_columns].astype(str)
+            df[string_columns] = df[string_columns].mask(
+                lambda x: x.notna(), df[string_columns].astype(str)  # type: ignore
```

Comment:
what does this do? specifically the mask part

Reply:
is this saying, if x is not na, convert it to a string?

Reply:
It's an if-else construct - if not NA, assign astype(str), else keep existing value.

There's probably a more idiomatic way to do that in pandas.

Reply:
Yeah I wondered the same and I believe that is it. It might help to use kwargs here to make it easier to understand:
`mask(cond=lambda x: x.notna(), other=df[string_columns].astype(str))`

Reply:
is this call vectorized?